File: Examples/ShareTradingExample_Java/src/main/java/dojo/sharetrading/domain/BuyRequest.java
package dojo.sharetrading.domain;

public class BuyRequest {

    private String accountNumber;
    private String productCode;

    public String getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(String accountNumber) {
        this.accountNumber = accountNumber;
    }

    public String getProductCode() {
        return productCode;
    }

    public void setProductCode(String productCode) {
        this.productCode = productCode;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public int getMaxUnitPrice() {
        return maxUnitPrice;
    }

    public void setMaxUnitPrice(int maxUnitPrice) {
        this.maxUnitPrice = maxUnitPrice;
    }

    private int quantity;
    private int maxUnitPrice;

    public BuyRequest() {
        super();
    }

    public BuyRequest(String AccountNumber, String ProductCode, int Quantity, int MaxUnitPrice)
    {
        accountNumber = AccountNumber;
        productCode = ProductCode;
        quantity= Quantity;
        maxUnitPrice = MaxUnitPrice;
    }
}

File: Examples/ShareTradingExample_CSharp/Models/BuyRequest.cs
﻿using System;
using System.ComponentModel.DataAnnotations;

namespace ShareTradingAPI.ViewModels
{
    public class BuyRequest
    {
        public Guid AccountNumber { get; set; }

        public string ProductCode { get; set; }

        [Range(0, int.MaxValue)]
        public int Quantity { get; set; }

        [Range(0, int.MaxValue)]
        public int MaxUnitPrice { get; set; }
    }
}
